sensor/common/metrics/metrics.go

```diff
@@ -432,3 +472,54 @@ func SetTelemetryMetrics(clusterIDPeeker func() string, cm *central.ClusterMetri
 	telemetrySecuredVCPU.Reset()
 	telemetrySecuredVCPU.WithLabelValues(labels...).Set(float64(cm.GetCpuCapacity()))
 }
+
+// ObserveCentralReceiverProcessMessageDuration records the duration of a ProcessMessage call
+func ObserveCentralReceiverProcessMessageDuration(componentName string, duration time.Duration) {
+	componentProcessMessageDurationSeconds.With(prometheus.Labels{
+		ComponentName: componentName,
+	}).Observe(duration.Seconds())
+}
+
+// IncrementCentralReceiverProcessMessageErrors increments the error count for a component's ProcessMessage call
+func IncrementCentralReceiverProcessMessageErrors(componentName string) {
+	componentProcessMessageErrorsCount.With(prometheus.Labels{
+		ComponentName: componentName,
+	}).Inc()
+}
+
+// GetMetricValue retrieves the current value of a prometheus metric with the given labels
+func GetMetricValue(t *testing.T, metricName string, labels map[string]string) float64 {
+	families, err := prometheus.DefaultGatherer.Gather()
+	assert.NoError(t, err)
+
+	for _, family := range families {
+		if family.GetName() == metricName {
+			for _, metric := range family.GetMetric() {
+				labelMatch := true
+				for requiredKey, requiredValue := range labels {
+					found := false
+					for _, labelPair := range metric.GetLabel() {
+						if labelPair.GetName() == requiredKey && labelPair.GetValue() == requiredValue {
+							found = true
+							break
+						}
+					}
+					if !found {
+						labelMatch = false
+						break
+					}
+				}
+
+				if labelMatch {
+					if metric.GetCounter() != nil {
+						return metric.GetCounter().GetValue()
+					}
+					if metric.GetGauge() != nil {
+						return metric.GetGauge().GetValue()
+					}
+				}
+			}
+		}
+	}
+	return 0
+}
```

Comment:
I see there were some changes recently. I think this should be in the testing file, not in the production code. I would suggest to have the metrics exported and move this func to the `central_receiver_test.go`.

Reply:
well, then into `testing_test.go` with it :)

Reply:
I agree, I feel this should be in a `_test.go` file.